Model/ProductProtection.php

```diff
@@ -7,94 +7,234 @@
+                $quote->removeItem($cartItemId);
+                $this->quoteRepository->save($quote);
+                return;
+            }
+
+            if (isset($cartItemId)) {
+                $item = $this->checkoutSession->getQuote()->getItemById($cartItemId);
+            }
+
+            // if we are adding pp, or we didn't find an existing item, create a new one
+            if (!isset($item) || $item === false) {
+                // ensure that we have the required properties to create the protection plan
+                if (
+                    !isset($quantity) ||
+                    !isset($productId) ||
+                    !isset($planId) ||
+                    !isset($price) ||
+                    !isset($term) ||
+                    !isset($coverageType)
+                ) {
+                    throw new LocalizedException(
+                        new Phrase('Missing required parameters to add product protection to cart.')
+                    );
+                }
+                $item = $this->itemFactory->create();
+            }
+
+            // if quote has not been saved yet, save it so that we have an id
+            if (!isset($quoteId)) {
+                $this->quoteRepository->save($quote);
+                $quote = $this->checkoutSession->getQuote();
+                $quoteId = $quote->getId();
+            }
+
+            $item->setQuoteId($quoteId);
+            $item->setProduct($product);
+
+            if (isset($quantity)) {
+                $item->setQty($quantity);
+            }
+
+            if (isset($price)) {
+                $item
+                    ->setCustomPrice($price / 100)
+                    ->setOriginalCustomPrice($price / 100)
+                    ->getProduct()
+                    ->setIsSuperMode(true);
+            }
+
+            $options = $this->createOptions($product, $item, [
+                'Plan Type' => $coverageType,
+                'Term' => $term,
+                'List Price' => $listPrice,
+                'Order Offer Plan Id' => $orderOfferPlanId,
+                'Lead Token' => $leadToken,
+                'Associated Product' => $productId,
+                'Plan ID' => $planId,
+            ]);
+            $item->setOptions($options);
+            $quote->addItem($item);
+
+            $this->quoteRepository->save($quote);
+            $quote->collectTotals();
+        } catch (Exception | LocalizedException $exception) {
+            $this->logger->error(
+                'Extend Product Protection Upsert Encountered the Following Exception ' .
+                    $exception->getMessage()
+            );
+            $this->integration->logErrorToLoggingService(
+                $exception->getMessage(),
+                $this->storeManager->getStore()->getId(),
+                'error'
+            );
+            // this is handled by magento error handler
+            throw $exception;
```

Comment:
LocalizedExceptions throw a 400, and other exceptions are handled by magentos Error Handler. I've seen 404 and 500 so far

Reply:
The only thing is this is not logging the message in the console. I think that should be okay, because we log the message using magento node-core, and tracking in the SDK can log the plan, storeid, etc.. of the plan that failed? Thoughts here?

Reply:
It is masked by magento that's why it does not get logged